nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/main/java/org/nuxeo/ecm/platform/audit/service/DefaultAuditBulker.java

```diff
@@ -176,20 +165,21 @@ public void run() {
                     if (queue.isEmpty()) {
                         continue;
                     }
+                    try {
+                        int count = drain();
+                        if (log.isDebugEnabled()) {
+                            log.debug("flushed " + count + " events");
+                        }
+                    } catch (RuntimeException cause) {
+                        log.error("caught error while draining audit queue", cause);
```

Comment:
What kind of exceptions are we expecting here? Shouldn't the queue be forcibly cleared if `drain` throws an exception so that later invariants (isEmpty) are still true?